src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -261,59 +281,121 @@ private BoundExpression BindCompoundAssignment(AssignmentExpressionSyntax node,
                 {
-                    checkedName = OperatorFacts.CompoundAssignmentOperatorNameFromSyntaxKind(operatorToken.Kind(), isChecked: true);
+                    checkedName = OperatorFacts.CompoundAssignmentOperatorNameFromSyntaxKind(node.OperatorToken.Kind(), isChecked: true);
                     if (!SyntaxFacts.IsCheckedOperator(checkedName))
                     {
                         checkedName = null;
                     }
                 }
 
+                ordinaryName = OperatorFacts.CompoundAssignmentOperatorNameFromSyntaxKind(node.OperatorToken.Kind(), isChecked: false);
+
+                return true;
+            }
+
+            BoundCompoundAssignmentOperator? tryApplyUserDefinedInstanceOperator(
+                AssignmentExpressionSyntax node,
+                BinaryOperatorKind kind,
+                bool checkOverflowAtRuntime,
+                string? checkedName,
+                string ordinaryName,
+                BoundExpression left,
+                BoundExpression right,
+                BindingDiagnosticBag diagnostics)
+            {
+                var leftType = left.Type;
+                Debug.Assert(leftType is not null);
+
+                if (leftType.SpecialType.IsNumericType())
+                {
+                    return null;
+                }
+
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+
                 ArrayBuilder<MethodSymbol>? methods = LookupUserDefinedInstanceOperators(
                     leftType,
                     checkedName: checkedName,
-                    ordinaryName: OperatorFacts.CompoundAssignmentOperatorNameFromSyntaxKind(operatorToken.Kind(), isChecked: false),
+                    ordinaryName: ordinaryName,
                     parameterCount: 1,
                     ref useSiteInfo);
 
+                diagnostics.Add(node, useSiteInfo);
+
                 if (methods?.IsEmpty != false)
                 {
-                    diagnostics.Add(node, useSiteInfo);
                     methods?.Free();
                     return null;
                 }
 
+                AnalyzedArguments? analyzedArguments = null;
+
+                BoundCompoundAssignmentOperator? inPlaceResult = tryInstanceOperatorOverloadResolutionAndFreeMethods(node, kind, checkOverflowAtRuntime, left, right, ref analyzedArguments, methods, diagnostics);
+
+                Debug.Assert(analyzedArguments is not null);
+                analyzedArguments.Free();
+
+                return inPlaceResult;
+            }
+
+            BoundCompoundAssignmentOperator? tryInstanceOperatorOverloadResolutionAndFreeMethods(
+                AssignmentExpressionSyntax node,
+                BinaryOperatorKind kind,
+                bool checkOverflowAtRuntime,
+                BoundExpression left,
+                BoundExpression right,
+                ref AnalyzedArguments? analyzedArguments,
```

Comment:
It might be reasonable to make this an `out AnalyzedArguments analyzedArguments`, since it doesn't look like the value coming in is used, and then the assertion at the call site could be removed.